### ISSUE TITLE
DVD that can only be written once as storage medium

### ISSUE THREAD

**Issue by Golddouble** [borgbackup/borg#6256]
######This is a question

#### Describe the problem you're observing.
A: When I only save my backup on my external HD, it can be, that something happens with this HD and my backup is lost (maybe Solar rupture). So it is a good practice to save another backup on a storage medium with a different technology. (Please note: I do not want to save my backup in a cloud.)

#### How much data is handled by borg?
A: About 500 MB (so there is enough space on the DVD)

Question: Can I use a DVD or Blu-ray, that can only be written once, as a storage medium for my Vorta/Borg backups?

Of course I am aware, that in this case I can not delete old "incremental backups", because nothing could be deleted on such a medium. But when I never delete an "incremental backup" but only make a new "incremetnal backup" each week: Is it possible to use then a DVD or Blu-ray, that can only be written once?

So basically the question is: Does Borg have to modify existing backup files for each new "incremental backup"? Or do existing files remain unchanged and new ones are simply added?

Thank you.

**Comment by ThomasWaldmann** [comment]
I would never use such slow media for borg, rather get some more external HDDs and swap them regularly.

Besides the extreme slowness (esp. seek time for random access) and that you could not compact, there are also some other fundamental issues with write-once/read-only media, like the locks borg wants to create in the repo directory. And that the capacity is rather limited.

**Comment by jgrisham** [comment]
Perhaps the file data in the backup repository could be batch-exported to DVD/BD on a regular basis (e.g. and then the individual files replaced with symlinks); the DVD/BD might have to only then be available online during a restore-type operation?

Parts (i.e. metadata, cache, etc.) requiring frequent read-access would be kept on an actual HDD/SSD.

**Comment by jgrisham** [comment]
Perhaps the file data in the backup repository could be batch-exported to DVD on a regular basis (e.g. and then the individual files replaced with symlinks); the SVD might have to only then be available online during a restore-type operation?

**Comment by ThomasWaldmann** [comment]
if course you can always make a copy of your repo to somewhere else and in case of emergency copy it back from there to disk. but borg does not support having some stuff here (on disk) and some stuff there (on dvd), it all must be inside the repo directory and be r/w.

**Comment by Golddouble** [comment]
Following scenario:
I copy my repo to a write-once/read-only media.
I make the next "incremental backup" in the ordinary repository which in on my external HDD.

After, can I now just copy some new files and folders from my repository of my external HDD to the  write-once/read-only media. (Which ones would be a different question). 
Or should some files be replaced on the write-once/read-only media? 

If some files should be replaced, then this method would not work.

**Comment by ThomasWaldmann** [comment]
for a lot of files, borg does not modify them in-place, but adds new files and deletes old ones.
but for some, modification is also possible, e.g. if repo_dir/config is updated.

so, if you want to make a repo copy, it must be a precise 1:1 copy.
